packages/app-builder/src/models/testrun.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  type TestRunCreateInputDto,
+  type TestRunDto,
+  type TestRunStatusDto,
+  type TestRunDecisionDataDto,
+  type TestRunRuleExecutionDataDto,
+} from 'marble-api/generated/marblecore-api';
+import { knownOutcomes } from './outcome';
+import { toUpperCase } from 'remeda';
+
+type Outcome = (typeof knownOutcomes)[number];
+
+export type TestRunStatus = TestRunStatusDto;
+export const testRunStatuses: TestRunStatus[] = [
+  'up',
+  'down',
+  'pending',
+  'unknown',
+];
+
+export type TestRunRuleStatus = TestRunRuleExecutionDataDto['status'];
+export const testRunRuleStatuses: TestRunRuleStatus[] = [
+  'hit',
+  'no_hit',
+  'error',
+  'snoozed',
+];
+
+export interface TestRunRuleExecution {
```

Comment:
I'd be in favor of "TestRunRuleExecutionCount" or something like this (similar for TestRunDecision type below)